sections/futures/MarketInfoBox/MarketInfoBox.tsx

```diff
@@ -1,38 +1,125 @@
 import React from 'react';
 import styled from 'styled-components';
-import Wei from '@synthetixio/wei';
+import Wei, { wei } from '@synthetixio/wei';
 import InfoBox from 'components/InfoBox';
-import { formatCurrency, formatPercent } from 'utils/formatters/number';
-import { Synths } from '@synthetixio/contracts-interface';
+import { formatCurrency, formatPercent, zeroBN } from 'utils/formatters/number';
+import { CurrencyKey, Synths } from '@synthetixio/contracts-interface';
+import { PotentialTrade } from '../types';
+import PreviewArrow from 'components/PreviewArrow';
+import useGetFuturesPotentialTradeDetails from 'queries/futures/useGetFuturesPotentialTradeDetails';
+import { FuturesPosition, FuturesPotentialTradeDetails } from 'queries/futures/types';
 
 type MarketInfoBoxProps = {
-	totalMargin: Wei;
-	availableMargin: Wei;
-	buyingPower: Wei;
-	marginUsage: Wei;
+	position: FuturesPosition | null;
 	isMarketClosed: boolean;
+	potentialTrade: PotentialTrade | null;
+	marketMaxLeverage: Wei | undefined;
+	currencyKey: string;
 };
 
 const MarketInfoBox: React.FC<MarketInfoBoxProps> = ({
-	totalMargin,
-	availableMargin,
-	buyingPower,
-	marginUsage,
+	position,
 	isMarketClosed,
+	potentialTrade,
+	marketMaxLeverage,
+	currencyKey,
 }) => {
+	const totalMargin = position?.remainingMargin ?? zeroBN;
+	const availableMargin = position?.accessibleMargin ?? zeroBN;
+	const buyingPower =
+		position && position?.remainingMargin.gt(zeroBN)
+			? position?.remainingMargin?.mul(marketMaxLeverage ?? zeroBN)
+			: zeroBN;
```

Comment:
Need to merge in `v2-dev` and update this to match the buying power calculation changes